tests/e2e/test_notification_e2e_helpers.py

```
@@ -0,0 +1,141 @@
+    deadline = asyncio.get_running_loop().time() + 10.0
+
+    assert await helpers.retry_after_rate_limit(
+        response(429, headers={"retry-after": "not-a-number"}), deadline=deadline
+    )
+    assert sleeps == [1.0]
+
+
+async def test_rate_limit_retry_never_sleeps_past_polling_deadline(monkeypatch):
+    sleeps = []
+
+    async def capture_sleep(delay):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `capture_sleep`

Add return type annotation: `None`

(ANN202)